core/src/main/scala/cats/Foldable.scala

```diff
@@ -107,9 +107,9 @@ import simulacrum.typeclass
    * needed.
    */
   def traverse_[G[_], A, B](fa: F[A])(f: A => G[B])(implicit G: Applicative[G]): G[Unit] =
-    foldLeft(fa, G.pure(())) { (acc, a) =>
-      G.map2(acc, f(a)) { (_, _) => () }
-    }
+    foldRight(fa, Later(G.pure(()))) { (a, acc) =>
```

Comment:
You might want to use `Always` here, since it will be evaluated once and there's no need for memoization.